src/common/snippets/src/lowered/shape_inference/shape_inference.cpp

```diff
@@ -0,0 +1,86 @@
+}
+
+ShapeInferPtr IShapeInferSnippetsFactory::get_specific_op_shape_infer(const ov::DiscreteTypeInfo& key,
+                                                                      const std::shared_ptr<ov::Node>& op) const {
+    return {};
+}
+
+#define SHAPE_INFER_PREDEFINED(OP, InferType) \
+    { OP::get_type_info_static(), [](const std::shared_ptr<ov::Node>& n) { return std::make_shared<InferType>();} }
+#define SHAPE_INFER_OP_SPECIFIC(OP) \
+    { OP::get_type_info_static(), [](const std::shared_ptr<ov::Node>& n) { return std::make_shared<OP::ShapeInfer>(n);} }
+#define SHAPE_INFER_OP_SPECIFIC_EXTERNAL(OP, InferType) \
+    { OP::get_type_info_static(), [](const std::shared_ptr<ov::Node>& n) { return std::make_shared<InferType>(n);} }
+
+const IShapeInferSnippetsFactory::TRegistry IShapeInferSnippetsFactory::registry {
+        // todo: Parameter and Scalar should be handled separately, since they have no inputs, and infer can't be called
+        SHAPE_INFER_PREDEFINED(op::ConvertTruncation, entryFirstPassthrough),
+        SHAPE_INFER_PREDEFINED(op::ConvertSaturation, entryFirstPassthrough),
+        SHAPE_INFER_PREDEFINED(op::Load, entryFirstPassthrough),
+        SHAPE_INFER_PREDEFINED(op::Store, entryFirstPassthrough),
+        SHAPE_INFER_PREDEFINED(op::Buffer, entryFirstPassthrough),
+        SHAPE_INFER_PREDEFINED(op::Fill, entryFirstPassthrough),
+        SHAPE_INFER_PREDEFINED(ov::op::v0::Parameter, entryFirstPassthrough),
+        SHAPE_INFER_PREDEFINED(ov::op::v1::Softmax, entryFirstPassthrough),
+        SHAPE_INFER_PREDEFINED(ov::op::v8::Softmax, entryFirstPassthrough),
+        //
+        SHAPE_INFER_PREDEFINED(op::LoopBegin, entrySingleElement),
+        SHAPE_INFER_PREDEFINED(op::Scalar, entrySingleElement),
+        SHAPE_INFER_PREDEFINED(op::LoopEnd, entryEmpty),
+        SHAPE_INFER_PREDEFINED(op::Nop, entryEmpty),
+        SHAPE_INFER_OP_SPECIFIC_EXTERNAL(opset1::Select, SelectShapeInfer),
+        // Note that Result has no output PortConnectors, so the shape must be empty
+        SHAPE_INFER_PREDEFINED(ov::op::v0::Result, entryEmpty),
+        //
+        SHAPE_INFER_OP_SPECIFIC(op::LoadReshape),
+        SHAPE_INFER_OP_SPECIFIC(op::Brgemm),
+        SHAPE_INFER_OP_SPECIFIC(op::HorizonMax),
+        SHAPE_INFER_OP_SPECIFIC(op::HorizonSum),
+        SHAPE_INFER_OP_SPECIFIC(op::VectorBuffer),
+        SHAPE_INFER_OP_SPECIFIC(op::BroadcastLoad),
+        SHAPE_INFER_OP_SPECIFIC(op::BroadcastMove),
+};
+#undef SHAPE_INFER_OP_SPECIFIC
+#undef SHAPE_INFER_PREDEFINED
+
+std::shared_ptr<IShapeInferSnippets> make_shape_inference(const std::shared_ptr<ov::Node>& op,
+                                                          const std::shared_ptr<IShapeInferSnippetsFactory>& factory) {
+    if (!factory) {
+        return nullptr;
+    } else if (auto shape_infer = factory->make(op->get_type_info(), op)) {
+        return shape_infer;
+    } else if (ov::is_type<ov::op::util::UnaryElementwiseArithmetic>(op)) {
```

Comment:
I don't think it's possible since it's one-to-one mapping, not dynamic_cast. Note that the same approach is used in [intel_cpu shapeInfer](https://github.com/openvinotoolkit/openvino/blob/d445f9ec95f75c58a52306cb9cf11ba48ef09700/src/plugins/intel_cpu/src/shape_inference/shape_inference.cpp#L558-L570), it's just a shortcut to avoid listing tons of operations.

Reply:
It's so sad! 😞 Thank you for the explanation!